include/zephyr/net/ieee802154_radio.h

```diff
@@ -757,61 +757,171 @@ enum ieee802154_config_type {
-	 * 3. Use @ref IEEE802154_CONFIG_CSL_PERIOD on each value change to update the current
-	 *    CSL period value which will be injected in the CSL IEs together with the CSL phase
-	 *    based on @ref IEEE802154_CONFIG_CSL_RX_TIME.
-	 * 4. Use @ref IEEE802154_CONFIG_RX_SLOT periodically to schedule the immediate receive
-	 *    window early enough before the expected window start time, taking into account
-	 *    possible clock drifts and scheduling uncertainties.
-	 *
-	 * This diagram shows the usage of the four options over time:
-	 *
-	 *         Start CSL                                  Schedule CSL window
-	 *
-	 *     ENH_ACK_HEADER_IE                        CSL_RX_TIME (following window)
-	 *          |                                        |
-	 *          | CSL_RX_TIME (nearest window)           | RX_SLOT (nearest window)
-	 *          |    |                                   |   |
-	 *          |    | CSL_PERIOD                        |   |
-	 *          |    |    |                              |   |
-	 *          v    v    v                              v   v
-	 *     ----------------------------------------------------------[ CSL window ]-----+
-	 *                                             ^                                    |
-	 *                                             |                                    |
-	 *                                             +--------------------- loop ---------+
+	 * Enables or disables a device as a CSL receiver and configures its CSL
+	 * period.
+	 *
+	 * @details Configures the CSL period in units of 10 symbol periods.
+	 * Values greater than zero enable CSL if the driver supports it and the
+	 * device starts to operate as a CSL receiver. Setting this to zero
+	 * disables CSL on the device. If the driver does not support CSL, the
+	 * configuration call SHALL return -ENOTSUP.
+	 *
+	 * See section 7.4.2.3 and section 8.4.3.6, table 8-104, macCslPeriod.
+	 *
+	 * @note Confusingly the standard calls the CSL receiver "CSL
+	 * coordinator" (i.e. "coordinating the CSL protocol timing", see
+	 * section 6.12.2.2), although, typically, a CSL coordinator is NOT also
+	 * an IEEE 802.15.4 FFD coordinator or PAN coordintor but a simple RFD
+	 * end device (compare the device roles outlined in sections 5.1, 5.3,
+	 * 5.5 and 6.1). To avoid confusion we therefore prefer calling CSL
+	 * coordinators (typically an RFD end device) "CSL receivers" and CSL
+	 * peer devices (typically FFD coordinators or PAN coordinators) "CSL
+	 * transmitters". Also note that at this time, we do NOT support
+	 * unsynchronized transmission with CSL wake up frames as specified in
+	 * section 6.12.2.4.4.
+	 *
+	 * To offload CSL receiver timing to the driver the upper layer SHALL
+	 * combine several configuration options in the following way:
+	 *
+	 * 1. Use @ref IEEE802154_CONFIG_ENH_ACK_HEADER_IE once with an
+	 *    appropriate pre-filled CSL IE and the CSL phase set to zero.  The
```

Comment:
Is it an actual requirement that the CSL Phase is set to zero when configuring `IEEE802154_CONFIG_ENH_ACK_HEADER_IE`? If the driver injects the CSL Phase on-the-fly, its value in the provided IE is ignored anyway.